src/user-attribute-handler.js

```diff
@@ -3,20 +3,24 @@ function UserAttributeHandler(common) {
 }
 
 UserAttributeHandler.prototype.onRemoveUserAttribute = function(key) {
-    var attribute = {};
-    attribute[key] = null;
-    window['optimizely'].push({
-        type: 'user',
-        attributes: attribute
-    });
+    if (window.optimizely) {
+        var attribute = {};
+        attribute[key] = null;
+        window['optimizely'].push({
+            type: 'user',
+            attributes: attribute
+        });
+    }
 };
 UserAttributeHandler.prototype.onSetUserAttribute = function(key, value) {
-    var attribute = {};
-    attribute[key] = value;
-    window['optimizely'].push({
-        type: 'user',
-        attributes: attribute
-    });
+    if (window.optimizely) {
+        var attribute = {};
+        attribute[key] = value;
+        window['optimizely'].push({
+            type: 'user',
+            attributes: attribute
+        });
+    }
```

Comment:
the above looks good.

the one thing that is missing is the part for optimizely FS, which involves requiring `common`, and then modifying it on both of these methods here. lmk if you have questions about that.